data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/resync/ResyncWorker.java

```diff
@@ -0,0 +1,131 @@
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.model.opensearch.OpenSearchBulkActions;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.source.rds.RdsSourceConfig;
+import org.opensearch.dataprepper.plugins.source.rds.converter.RecordConverter;
+import org.opensearch.dataprepper.plugins.source.rds.coordination.partition.ResyncPartition;
+import org.opensearch.dataprepper.plugins.source.rds.coordination.state.ResyncProgressState;
+import org.opensearch.dataprepper.plugins.source.rds.schema.QueryManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+
+public class ResyncWorker implements Runnable {
+    private static final Logger LOG = LoggerFactory.getLogger(ResyncWorker.class);
+
+    static final String DATA_PREPPER_EVENT_TYPE = "event";
+    static final Duration BUFFER_TIMEOUT = Duration.ofSeconds(60);
+    static final int DEFAULT_BUFFER_BATCH_SIZE = 1_000;
+
+    private final ResyncPartition resyncPartition;
+    private final RdsSourceConfig sourceConfig;
+    private final QueryManager queryManager;
+    private final Buffer<Record<Event>> buffer;
+    private final RecordConverter recordConverter;
+    private final AcknowledgementSet acknowledgementSet;
+
+    ResyncWorker(ResyncPartition resyncPartition,
+                 RdsSourceConfig sourceConfig,
+                 QueryManager queryManager,
+                 Buffer<Record<Event>> buffer,
+                 RecordConverter recordConverter,
+                 AcknowledgementSet acknowledgementSet) {
+        this.resyncPartition = resyncPartition;
+        this.sourceConfig = sourceConfig;
+        this.queryManager = queryManager;
+        this.buffer = buffer;
+        this.recordConverter = recordConverter;
+        this.acknowledgementSet = acknowledgementSet;
+    }
+
+    public static ResyncWorker create(ResyncPartition resyncPartition,
+                                      RdsSourceConfig sourceConfig,
+                                      QueryManager queryManager,
+                                      Buffer<Record<Event>> buffer,
+                                      RecordConverter recordConverter,
+                                      AcknowledgementSet acknowledgementSet) {
+        return new ResyncWorker(resyncPartition, sourceConfig, queryManager, buffer, recordConverter, acknowledgementSet);
+    }
+
+    public void run() {
+        String[] keySplits = resyncPartition.getPartitionKey().split("\\|");
+        final String database = keySplits[0];
+        final String table = keySplits[1];
+        final long eventTimestampMillis = Long.parseLong(keySplits[2]);
+
+        if (resyncPartition.getProgressState().isEmpty()) {
```

Comment:
Move to separate method to validate and get progress state

```
final ResyncProgressState progressState = validateAndGetProgressState();
```

```
private ResyncProgressState validateAndGetProgressState() {
    return resyncPartition.getProgressState()
            .orElseThrow(() -> new IllegalStateException(
                    "ResyncPartition " + resyncPartition.getPartitionKey() + " doesn't contain progress state."));
}
```

Reply:
Done.